### DISCUSSION TITLE
Logging of Extrx values 0

### DISCUSSION THREAD

**Discussion by vBruegge** [bitcraze/discussions#1140] [Q&A]
Hello,
yesterday I tried to log data via the cfclient and encountered the problem that all values could be logged except the value in extrx and extrx_raw. These remained 0 even while the drone was flying the send commands. Since I  was using a separate RC, I thought that that might be the problem and connected a gamepad to the laptop running the cfclient. However, also this didn't do the trick and my extrx values remained 0. How can I save these values or are there any other logged variables which I can use instead?
Thanks in advance

**Comment by knmcguire** [comment]
Hi!

I think these values are used for external receiver in put of 'stand alone' transmitter/controllers, so not through the cfclient. The crtp packet for controllers from the CFclient gets decoded here: https://github.com/bitcraze/crazyflie-firmware/blob/master/src/modules/src/crtp_commander_rpyt.c#L120

Here you can see that the setpoints get overwritten there, so you should probably look at the controller logging group instead: https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/api/logs/#controller

**Reply by vBruegge** [reply]
Hi,

I controlled the drone most of the time with an external RC, and used the cfclient just for the logging of the parameters. So both, the client and the remote were connected to the drone, I sent commands via remote and received the logged values for extrx, the stabilizer... in the client.
With this setup, my values for extrx remained 0. This was the reason why I tried controlling the drone with the cfclient.
Hope that clarifies my issue.
Thanks for the help!

**Reply by tobbeanton** [reply]
Follow up question. What type of external RC controller and receiver did you use? The extrx log variables will be shown for controllers connected to the CPPM firmware module, usually CPPM signal from the receiver to the BigQuad-deck.

**Reply by amitx64** [reply]
@tobbeanton Similarly, I am trying to `log the arming status`, but it always shows zero. I attempted `arming` and `disarming` using `cfclient`, but I consistently receive a value of 0 even when I send the arm command through cfclient.

Here is my code used to log the arming status:
```
#include "config.h"
#include "debug.h"
#include "FreeRTOS.h"
#include "queue.h"
#include "static_mem.h"
#include "task.h"
#include "log.h"

static bool isInit = false;
static uint8_t arm_status = 0;
static logVarId_t logIdARM;

static uint8_t get_ARM_STATUS() { return logGetInt(logIdARM); }

static void exampleTask(void*);
STATIC_MEM_TASK_ALLOC(exampleTask, SENSOR_VALUE_TASK_STACKSIZE);

void exampleTaskInit() {
  // Get log and param ids
  logIdARM = logGetVarId("extrx", "Arm");
  STATIC_MEM_TASK_CREATE(exampleTask, exampleTask, SENSOR_VALUE_TASK_NAME, NULL, SENSOR_VALUE_TASK_PRI);
  isInit = true;
}

bool exampleTaskTest() {
  return isInit;
}

static void exampleTask(void* parameters) {
  DEBUG_PRINT("Example task main function is running!");
  while (true) {
    vTaskDelay(M2T(20));
    arm_status = get_ARM_STATUS();

    if(arm_status == 1)
    {
        DEBUG_PRINT("ARM\n");
    }
    else{
        DEBUG_PRINT("DISARM\n");
    }
  } 
}
```

**Reply by knmcguire** [reply]
@amitx64 This is not the same situation as the original poster is having, so please start a new discussion.

Just mind that I've explained this a couple of times before, so please mind the discussion rules: https://github.com/orgs/bitcraze/discussions/55. Unless you are sure that you are experiencing the same issue as the one that started the discussion, always start a new discussion, as this will complicate it for the original poster's support issue. We want to avoid removing posts because of this reason.

**Reply by amitx64** [reply]
Ok
